### PR TITLE
Palworld: Explain updating and the parser in the README

### LINKED ISSUE #2715: [Egg Request]: Palworld How Do I set the all game configuration on the panel
### Does this expand an already existing service

Yes

### Link to a game or software

https://github.com/parkervcp/eggs/tree/master/game_eggs/steamcmd_servers/palworld

### Links for the download

https://store.steampowered.com/app/1623730/Palworld/

### Links for the install docs

https://tech.palworldgame.com/dedicated-server-guide

### PR DESCRIPTION
**Pull request by QuintenQVD0**
# Description

- Explain how this egg should be updated if a server was already made
- update + explain the extra variables and that if needed hosts can add variables themselves as the parser can now handle them

closes #2715 
## Checklist for all submissions

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you tested and reviewed your changes with confidence that everything works?
* [x] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

### CODE REVIEW

File: game_eggs/steamcmd_servers/palworld/README.md [hunk 1]
```diff
@@ -98,4 +98,21 @@ As of January 19th 2024 the server requires about 5GB of storage capacity. This
 | Game            | 8211    |
 | RCON (optional) | 25575   |
 
-The RCON port does not need to be allocated.
\ No newline at end of file
+The RCON port does not need to be allocated.
+
+### Updating
```
Comment (parkervcp) [inline game_eggs/steamcmd_servers/palworld/README.md:103]: We could make this a standalone doc in the repo named `UPDATING_EGGS.md` or something.

The parser docs are fine to keep.

**Review by parkervcp — CHANGES_REQUESTED** [review]

File: game_eggs/steamcmd_servers/palworld/README.md (hunk 1 above)
Comment (lilkingjr1) [reply game_eggs/steamcmd_servers/palworld/README.md:103]: I second this.

Comment (schlags) [reply game_eggs/steamcmd_servers/palworld/README.md:103]: I'm not sure if it makes sense to new users to simply say "Update your egg". Does this mean creating a new egg in the nest?

File: game_eggs/steamcmd_servers/palworld/README.md [hunk 2]
```diff
@@ -98,4 +98,21 @@ As of January 19th 2024 the server requires about 5GB of storage capacity. This
...
+
+1. Update your egg
+2. Update the startup of all already made servers to the one that now comes with the egg
+3. Then hit reinstall because the parser application has to be downloaded
```
Comment (schlags) [inline game_eggs/steamcmd_servers/palworld/README.md:107]: Also worth calling out if saves will be wiped here, and if how to restore them - or if reinstall is simply for config/images/binaries and not the mounted directory for the saves

**Comment by LAIguapi** [comment]
I am very happy that the questions I have raised can be valued by you. At the same time, I am very interested in the design of the Pterodactyl. I do n’t know if you can help me answer my little questions about the it.

1. The panel is to load the startup script first, and then load the docker script, right?
2. Can I save the DockerFile to the local, thereby reducing the time of the server from remotely

Looking forward to getting your answer

**Comment by schlags** [comment]
```json
        {
            "name": "Death Penalty",
            "description": "0 drop nothing\r\n1 enables the dropping of only items\r\n2 to drop items and equipment\r\n3 to drop items, equipment, and pals in your current party.",
            "env_variable": "DEATH_PENALTY",
            "default_value": "",
            "user_viewable": true,
            "user_editable": true,
            "rules": "required|numeric|in:0,1,2,3",
            "field_type": "text"
        },
```
This might be a good one to add to the default startup settings

**Comment by parkervcp** [comment]
> I am very happy that the questions I have raised can be valued by you. At the same time, I am very interested in the design of the Pterodactyl. I do n’t know if you can help me answer my little questions about the it.
> 
>     1. The panel is to load the startup script first, and then load the docker script, right?
> 
>     2. Can I save the DockerFile to the local, thereby reducing the time of the server from remotely
> 
> 
> Looking forward to getting your answer

This unrelated to this pr.

**Comment by Opalatal** [comment]
Howdy, do we have an ETA for this fork?

Currently unable to edit certain server config settings.

**Comment by QuintenQVD0** [comment]
> Howdy, do we have an ETA for this fork?
> 
> Currently unable to edit certain server config settings.

The current egg only allow for the basis. This pr will allow hosting to add extra variables.

**Comment by QuintenQVD0** [comment]
I just updated this PR to the latest version of the parser application (We will probably soon move to downloading directly from parker fork)

### Changes
- There now is a version print
- The Region Key was added
- If a `DefaultPalWorldSettings.ini` exists but a `PalWorldSettings.ini` does not, it will try to copy it to the right directory.
- If a `DefaultPalWorldSettings.ini` exists and a `PalWorldSettings.ini` exits, but it is empty, then it will try to copy the contents over to the settings file.
- If you want to clear / empty a value in the config then you must set the ENV variable value to `nil`or `null`, putting just an empty variable will not edit the value in the config.

### Validation
There is some very basic validation on the variables. (Those are build in to the parser and can not be changed, things like length validation must still be done through the panel)

|Rule|Value|Example|
|-----|-----|-----|
|Numeric|Allows only positive numeric values|"123" or "25565"|
|Floating|Allows only positive floating-point values|"0.005" or "3.14"|
|TrueFalse| Allows values "True" or "False"|True" or "False"|
|String|Everything|"this is a test" or "test"|
|AlphaDash|Allows only alphanumeric characters and dashes|"abc123" or"test-123"|

**Review by parkervcp — APPROVED** [review]

**Pull request merged**